vm/rust/src/lib.rs

```diff
@@ -166,6 +168,36 @@ pub extern "C" fn cairoVMExecute(
                     maybe_cc = contract_class_from_sierra_json(class_json_str.get())
                 };
 
+                let class_hash = match serde_json::from_str(class_json_str.get()).unwrap() {
+                    ContractArtifact::LegacyClass(c) =>  starknet_api::core::ClassHash(starknet_api::hash::StarkFelt::new(c.class_hash().unwrap().to_bytes_be()).unwrap()),
+                    ContractArtifact::SierraClass(c) =>  starknet_api::core::ClassHash(starknet_api::hash::StarkFelt::new(c.class_hash().unwrap().to_bytes_be()).unwrap()),
+                    ContractArtifact::CompiledClass(c) =>  starknet_api::core::ClassHash(starknet_api::hash::StarkFelt::new(c.class_hash().unwrap().to_bytes_be()).unwrap()),
+                };
+
+                sn_api_txn = match declare {
+                    starknet_api::transaction::DeclareTransaction::V0(_) => {
+                        starknet_api::transaction::Transaction::Declare(starknet_api::transaction::DeclareTransaction::V0(starknet_api::transaction::DeclareTransactionV0V1 {
+                            max_fee: declare.max_fee(),
+                            transaction_hash: declare.transaction_hash(),
+                            signature: declare.signature(),
+                            nonce: declare.nonce(),
+                            sender_address: declare.sender_address(),
+                            class_hash,
+                        }))
+                    },
+                    starknet_api::transaction::DeclareTransaction::V1(_) => {
+                        starknet_api::transaction::Transaction::Declare(starknet_api::transaction::DeclareTransaction::V1(starknet_api::transaction::DeclareTransactionV0V1 {
+                            max_fee: declare.max_fee(),
+                            transaction_hash: declare.transaction_hash(),
+                            signature: declare.signature(),
+                            nonce: declare.nonce(),
+                            sender_address: declare.sender_address(),
+                            class_hash,
+                        }))
+                    },
+                    _ => sn_api_txn,
+                };
+
```

Comment:
We need class hash to be able to calculate the transaction hash. Which we do on Go side. So Go should be able to calculate the class hash before calling cairoVmExecute.

See how I wrapped the rust code to be called from Go 

https://github.com/NethermindEth/juno/pull/809/commits/a5a6edaef57678f83ebc3479ca94f545f6bd7534

and how it is used

https://github.com/NethermindEth/juno/pull/809/commits/a6f60267fb82e0c34fd21968bdb754bd16e41c63#diff-14aba5c63ad044ac1e8a62d96fed2cd132ad5e4935fc555a935a40c14f5cb0f7R156-R169